storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -521,10 +524,11 @@ UInt64 getHighestVotedValidatorIndex() {
     }
   }
 
-  VoteTracker getVote(UInt64 validatorIndex) {
+  Optional<VoteTracker> getVote(UInt64 validatorIndex) {
```

Comment:
While we do tend to prefer `Optional` over returning `null`, in this particular case we'll be iterating through 200k+ of these in quick succession and immediately throwing away the `Optional` which puts a lot of unnecessary pressure on the GC.  So I'd suggest going back to returning `null` here even though it does make `StoreVoteUpdater.getVote` slightly more complex.  Fortunately that's the only caller of this method so we aren't exposing the null pointer risk to the public API of `Store` and the rest of the application.

Reply:
done